compiler/daml-lf-verify/src/DA/Daml/LF/Verify/Context.hs

```diff
@@ -98,52 +118,91 @@ createCond :: BoolExpr
   -> Cond a
 createCond cond x y = Conditional cond [Determined x] [Determined y]
 
--- | Shift the conditional inside the update set.
-introCond :: Cond (UpdateSet ph) -> UpdateSet ph
+extCond :: Eq a => BoolExpr -> Cond a -> [Cond a]
+extCond bexp cond =
+  let cond' = case cond of
+        (Determined x) -> Conditional bexp [Determined x] []
+        (Conditional bexp' xs ys) -> Conditional (bexp `BAnd` bexp') xs ys
+  in simplifyCond cond'
+
+-- | Perform common simplifications on Conditionals.
```

Comment:
Nothing would break if we stopped performing these simplifications. The only reason why we have them, is to make the output look less daunting to the user.

Reply:
Thanks! Would be great to add a comment for this.